File: src/main/java/com/example/liftoff/storage/web/DropboxStorage.java
package com.example.liftoff.storage.web;

import com.dropbox.core.DbxRequestConfig;
import com.dropbox.core.v2.DbxClientV2;
import com.dropbox.core.v2.files.WriteMode;
import com.example.liftoff.storage.Storage;

import java.io.InputStream;
import java.nio.file.Path;
import java.util.Optional;

public class DropboxStorage implements Storage {
    private final Path rootDirectory;
    private final String accessToken;

    public DropboxStorage(Path rootDirectory, String accessToken) {
        this.rootDirectory = rootDirectory;
        this.accessToken = accessToken;
    }

    @Override
    public void storeFile(InputStream file, String fileName, Optional<String> fileDirectoryName) throws Exception {
        // TODO implement and test Dropbox storage
        final var config = DbxRequestConfig.newBuilder("test").build();
        DbxClientV2 client = new DbxClientV2(config, accessToken);
        client.files().uploadBuilder(this.rootDirectory.toString()).withMode(WriteMode.OVERWRITE).uploadAndFinish(file);
    }

}

File: src/main/scala/com/example/liftoff/storage/web/DropboxStorage.scala
package com.example.liftoff.storage.web

import com.example.liftoff.error.StorageError
import com.example.liftoff.storage.Storage
import zio.IO

import java.io.InputStream

class DropboxStorage extends Storage {
  override def storeFile(file: InputStream, fileName: String, fileDirectoryName: Option[String]): IO[StorageError, Unit] = ???
}
